src/Entities/ArmorStand.cpp

```diff
@@ -0,0 +1,461 @@
+
+#include "Globals.h"  // NOTE: MSVC stupidness requires this to be the same across all modules
+
+#include "ArmorStand.h"
+#include "Player.h"
+#include "../ClientHandle.h"
+#include "../Chunk.h"
+#include "../World.h"
+
+
+
+
+
+cArmorStand::cArmorStand(Vector3d a_Pos, double a_Yaw, short a_Size):
+	cEntity(etArmorStand, a_Pos.x, a_Pos.y, a_Pos.z, 0.25*a_Size, 0.9875*a_Size),  // a_Size = 0 : Marker ;  a_Size = 1 : Small ;  a_Size = 0 : Normal
+	m_CustomName(""),
+	m_CustomNameAlwaysVisible(false),
+	m_Size(a_Size),
+	m_HasArms(false),
+	m_HasBasePlate(true),
+	m_HeadRotation(0.0, 0.0, 0.0),
+	m_BodyRotation(0.0, 0.0, 0.0),
+	m_LeftArmRotation(-10.0, 0.0, -10.0),
+	m_RightArmRotation(-15.0, 0.0, 10.0),
+	m_LeftLegRotation(-1.0, 0.0, -1.0),
+	m_RightLegRotation(1.0, 0.0, 1.0),
+	m_LeftHand(cItem()),
+	m_RightHand(cItem()),
+	m_Helmet(cItem()),
+	m_ChestPlate(cItem()),
+	m_Leggings(cItem()),
+	m_Boots(cItem())
+{
+	SetYaw(a_Yaw);
+	SetMaxHealth(1);
+	SetHealth(1);
+}
+
+
+
+
+
+void cArmorStand::SpawnOn(cClientHandle & a_ClientHandle)  // Should got any rotation and should replace any of non-solid blocks like snow
+{
+	a_ClientHandle.SendSpawnObject(*this, 78, 0, static_cast<Byte>(GetYaw()), static_cast<Byte>(GetPitch()));
+	a_ClientHandle.SendEntityMetadata(*this);
+	a_ClientHandle.SendEntityEquipment(*this, 0, GetEquippedWeapon());
+	a_ClientHandle.SendEntityEquipment(*this, 1, GetEquippedBoots());
+	a_ClientHandle.SendEntityEquipment(*this, 2, GetEquippedLeggings());
+	a_ClientHandle.SendEntityEquipment(*this, 3, GetEquippedChestplate());
+	a_ClientHandle.SendEntityEquipment(*this, 4, GetEquippedHelmet());
+	a_ClientHandle.SendEntityEquipment(*this, 5, GetOffHandEquipedItem());  // For compatibility issues, left hand was added at the end and not at the ID 1 like wants the protocol since 1.9
+}
+
+
+
+
+
+void cArmorStand::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
+{
+	UNUSED(a_Dt);
+	UNUSED(a_Chunk);
+	// I don't know how and why but by default the client considere it as no gravity, so disable gravity on the server unless it fixed totally and configurable
+	// Gravity for such entity is not really important
+}
+
+
+
+
+
+void cArmorStand::KilledBy(TakeDamageInfo & a_TDI)
+{
+	// TODO: Replace harm by shaking effect when hurt
+
+	if (IsMarker())  // Disallow interaction with marker
+	{
+		return;
+	}
+
+	cItems Pickup;
+	if (!m_LeftHand.IsEmpty())
+	{
+		Pickup.push_back(m_LeftHand);
+		SetOffHandEquipedItem(cItem());
+	}
+	else if (!m_RightHand.IsEmpty())
```

Comment:
Why are these conditions all `else`? Surely the armour stand should drop all of its armour pieces, not just the first one.

Reply:
Yes, I don't know with which gameplay I hit the part of the armor stand to get it back individually, but you are right (tested in 1.13.2); All fall suddenly